### PR TITLE
Mengyuan Pull Request HW7

### PR DESCRIPTION
**Pull request by MengyuanZoe**
@hanghu @YangJiang89 @jhp312

### CODE REVIEW

File: HW7/Mengyuan_HW7.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import os
+
+import numpy as np
+import pandas as pd 
+import matplotlib.pyplot as plt
```
Comment (puleo-joe) [inline HW7/Mengyuan_HW7.py]: Both numpy and matplotlib are unused imports.  They can be deleted from the script.

File: HW7/Mengyuan_HW7.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import folium 
+from folium import plugins
+import webbrowser
+
+
+# Set global settings and macros.
+MAX_SHOW =10000
```
Comment (puleo-joe) [inline HW7/Mengyuan_HW7.py]: Add a space as follows: 
MAX_SHOW = 10000

File: HW7/Mengyuan_HW7.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import os
```
Comment (puleo-joe) [inline HW7/Mengyuan_HW7.py:7]: Add a docstring to the beginning of the script

File: HW7/Mengyuan_HW7.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+# Set global settings and macros.
+MAX_SHOW =10000
+HOUSE_URL = 'houses.html'
+HOUSE_HEAT_URL = "househeatmap.html"
+CRIME_HEAT_URL = "crimeheatmap.html"
+pd.set_option('display.max_columns', None) # To display all columns
+
+
+# Read in king county house data.
+data = pd.read_csv("../../Data/kc_house_data.csv", parse_dates = ['date'])
```
Comment (puleo-joe) [inline HW7/Mengyuan_HW7.py]: Remove whitespace in rename parse_date = ['date'] to parse_date=['date'] ...
Similar arguments should also remove whitespace

File: HW7/Mengyuan_HW7.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+data['zipcode'] = data['zipcode'].astype(str)
+
+## Create one map shwoing each listing of in house dataset and show in browser.
```
Comment (puleo-joe) [inline HW7/Mengyuan_HW7.py]: spelling correction:  change "shwoing" to "showing"

File: HW7/Mengyuan_HW7.py (hunk 3 above)
Comment (hanghu) [reply HW7/Mengyuan_HW7.py:7]: module name should all be in lowercase.
Quote from PEP 8:
"Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged."

File: HW7/Mengyuan_HW7.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+#	When mouse-over the cluster, show house listing information:
+#	sqft, price.
+for iters, row in data[0:MAX_SHOW].iterrows():
+    folium.Marker([row["lat"], row["long"]], 
+                  popup="{0} sqft:  Sold for $ {1}"\
+                  .format(row["sqft_living"], row["price"])).add_to(marker_cluster)
+
+# Save the house cluster map to a html and open it in broswer.
+houses_map.save(HOUSE_URL)
+webbrowser.open('file://' + os.path.realpath(HOUSE_URL),new=2)
```
Comment (hanghu) [inline HW7/Mengyuan_HW7.py]: there usually is one space after comma

File: HW7/Mengyuan_HW7.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import os
+
+import numpy as np
+import pandas as pd 
+import matplotlib.pyplot as plt
+import folium 
+from folium import plugins
+import webbrowser
```
Comment (hanghu) [inline HW7/Mengyuan_HW7.py]: the standard import should be put in the first

File: HW7/Mengyuan_HW7.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+## Create one map showing the frequency of house sales and show in browser.
+
+# Use folium Map function to plot underlying basic map.
+houses_heatmap = folium.Map(location = [data['lat'].mean(), data['long'].mean()], zoom_start = 10)
+# Add heatmap on top of the basic map.
+houses_heatmap.add_children(
+        plugins.HeatMap([[row["lat"], row["long"],row["price"]] 
+            for iters, row in data.iterrows()],
+            	min_opacity=0.5, max_zoom=18, radius = 8)) #[0:MAX_SHOW]
```
Comment (hanghu) [inline HW7/Mengyuan_HW7.py]: The about two lines use wrong continued indentation,

File: HW7/Mengyuan_HW7.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+# Save the house sale frequency heat map to a html and open it in broswer.
+houses_heatmap.save(HOUSE_HEAT_URL)
+webbrowser.open('file://' + os.path.realpath(HOUSE_HEAT_URL),new=2)
+
+
+
```
Comment (hanghu) [inline HW7/Mengyuan_HW7.py]: The last couple of empty lines are redundant

**Pull request merged**